samza-kv-rocksdb/src/main/scala/org/apache/samza/storage/kv/RocksDbKeyValueStore.scala

```diff
@@ -188,7 +189,7 @@ class RocksDbKeyValueStore(
 
   def delete(key: Array[Byte]): Unit = ifOpen {
     metrics.deletes.inc
-    put(key, null)
```

Comment:
This prevents deletes from incrementing put metrics.

Reply:
Either way is fine. One way to interpret these metrics is as call counts to put/delete methods (this PR). Another is to interpret them as DB operations (your suggestion). I can update them to be db operations to be consistent with putAll (it increments puts and deletes as well).